rocAL/rocAL/include/pipeline/tensor.h

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright (c) 2019 - 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <VX/vx.h>
+#include <VX/vx_types.h>
+#include <vector>
+#include <cstring>
+#include <array>
+#include <queue>
+#include <memory>
+#if ENABLE_HIP
+#include "hip/hip_runtime.h"
+#include "device_manager_hip.h"
+#else
+#include "device_manager.h"
+#endif
+#include "commons.h"
+
+/*! \brief Converts Rocal Memory type to OpenVX memory type
+ *
+ * @param mem input Rocal type
+ * @return the OpenVX type associated with input argument
+ */
+vx_enum vx_mem_type(RocalMemType mem);
+
+vx_size tensor_data_size(RocalTensorDataType data_type);
+
+/*! \brief Holds the information about a rocALTensor */
+class rocALTensorInfo
+{
+public:
+    friend class rocALTensor;
+    enum class Type
+    {
+        UNKNOWN = -1,
+        REGULAR = 0,
+        VIRTUAL = 1,
+        HANDLE  = 2
+    };
+    
+    // Default constructor
+    /*! initializes memory type to host and dimension as empty vector*/
+    rocALTensorInfo();
+
+    //! Initializer constructor with only fields common to all types (Image/ Video / Audio)
+    rocALTensorInfo(std::vector<size_t> dims,
+                    RocalMemType mem_type,
+                    RocalTensorDataType data_type);
+
+    // Setting properties required for Image / Video
+    void set_roi_type(RocalROIType roi_type) { _roi_type = roi_type; }
+    void set_data_type(RocalTensorDataType data_type)
+    {
+        _data_type = data_type;
+        _data_size = (_data_size / _data_type_size);
+        _data_size *= data_type_size();
+    }
+    void set_max_dims()
```

Comment:
what is max_dims here? If this is width and height, please rename them as width and height

Reply:
In the case of image tensors we will store the max width and max height, but when it comes to audio tensors we will be storing the maximum frames and channels, hence we have kept a generic name as max_dims